primehub/datasets.py

```diff
@@ -0,0 +1,387 @@
+
+        :type src: str
+        :param src: the path of a local file or local directory
+
+        :type path: str
+        :param path: the path of the dataset
+
+        :type recursive: bool
+        :param recursive: copy recursively, set it when the source is a directory
+        """
+
+        self._check_dataset_exist(dataset_id)
+        full_path = gen_datasets_path(dataset_id, path)
+
+        try:
+            result = self.primehub.files.upload(src, full_path, **kwargs)
+            return result
+        except SharedFileException as e:
+            message = e.args[0]
+            message = message.replace(full_path, path)
+            raise SharedFileException(message)
+
+    @cmd(name='files-download', description='download files from the dataset', optionals=[('recursive', toggle_flag)])
+    def files_download(self, dataset_id: str, path: str, dest: str, **kwargs) -> dict:
+        """
+        Download files of the dataset by path
+
+        :type dataset_id: str
+        :param dataset_id: the name of the dataset
+
+        :type path: str
+        :param path: the path of a file or a directory
+
+        :type dest: str
+        :param dest: the local path to save files
+
+        :type recursive: bool
+        :param recursive: copy recursively, set it when the path is a directory
+        """
+
+        self._check_dataset_exist(dataset_id)
+        full_path = gen_datasets_path(dataset_id, path)
+
+        try:
+            result = self.primehub.files.download(full_path, dest, **kwargs)
+            return result
+        except SharedFileException as e:
+            message = e.args[0]
+            message = message.replace(full_path, path)
+            raise SharedFileException(message)
+
+    @cmd(name='files-delete', description='delete files from the dataset', optionals=[('recursive', toggle_flag)])
+    def files_delete(self, dataset_id: str, path: str, **kwargs) -> dict:
+        """
+        Delete a file or a directory of the dataset by path
+
+        :type dataset_id: str
+        :param dataset_id: the name of the dataset
+
+        :type path: str
+        :param path: the path of a file or a directory
+
+        :type recursive: bool
+        :param recursive: delete recursively, set it when the path is a directory
+        """
+
+        self._check_dataset_exist(dataset_id)
+        full_path = gen_datasets_path(dataset_id, path)
+
+        try:
+            result = self.primehub.files.delete(full_path, **kwargs)
+            return result
+        except SharedFileException as e:
+            message = e.args[0]
+            message = message.replace(full_path, path)
+            raise SharedFileException(message)
+
+    def _check_dataset_exist(self, dataset_id: str):
+        self.get(dataset_id)
+
+    def help_description(self):
+        return "Manage datasets"
+
+
+def validate_creation(config):
+    spec = ValidationSpec("""
+    input DatasetV2CreateInput {
+      id: String!
```

Comment:
The id should be unique and not contain segment chars `.` or `/`
Suggest to follow other id rules like k8s names
https://kubernetes.io/docs/concepts/overview/working-with-objects/names/#names

Reply:
ok, add a validator